### PR TITLE
Fix/core data concurrency

### PR DESCRIPTION
**Pull request by marcoconti83**
Fix core data concurrency issues and adopt new ZMTransport API (also modified for concurrency issues)

- [ ] https://github.com/wireapp/wire-ios-data-model/pull/113

### CODE REVIEW

File: Cartfile.resolved [hunk 1]
```diff
@@ -9,7 +9,7 @@ github "wireapp/wire-ios-link-preview" "4.0.2"
 github "wireapp/wire-ios-utilities" "13.0.1"
 github "wireapp/wire-ios-images" "7.0.0"
 github "wireapp/wire-ios-testing" "8.0.1"
-github "wireapp/wire-ios-transport" "16.0.0"
+github "wireapp/wire-ios-transport" "fea6ddb437dd8da10c34e6fdbb6f94ab28040164"
```
Comment (typfel) [inline Cartfile.resolved]: Need new version

Comment (marcoconti83) [inline Cartfile.resolved]: Need to bump everything actually, starting with this: 
https://github.com/wireapp/wire-ios-data-model/pull/113

File: Source/Calling/ZMGSMCallHandler.m [hunk 2]
```diff
@@ -179,22 +179,22 @@ - (void)rejoinVoiceChannelAfterGSMInterruption
 
 - (BOOL)hasStoredInterruptedCallConversation
 {
-    return ([self.uiManagedObjectContext persistentStoreMetadataForKey:ZMInterruptedCallConversationObjectIDKey] != nil);
+    return ([self.syncManagedObjectContext persistentStoreMetadataForKey:ZMInterruptedCallConversationObjectIDKey] != nil);
```
Comment (typfel) [inline Source/Calling/ZMGSMCallHandler.m]: So this should only be called from the sync context then? Which it's not I think.

Comment (marcoconti83) [inline Source/Calling/ZMGSMCallHandler.m]: I tried calling (outgoing/incoming) and I never hit the crash for CD concurrency here, so I guess it works.

Comment (marcoconti83) [inline Source/Calling/ZMGSMCallHandler.m]: ... as opposed to the other places where I had to make changes, those were making CD crash

Comment (marcoconti83) [inline Source/Calling/ZMGSMCallHandler.m]: Good point. Did not try that.

**Comment by marcoconti83** [comment]
@typfel please re-review

**Comment by marcoconti83** [comment]
I tested it and found issues in the code you were suspicious about :)
Had to add a lot of context switches but now it works

**Comment by typfel** [comment]
Nice

**Pull request merged**